### PR TITLE
Feat/atom button

### PR DESCRIPTION
**Pull request by emersonlaurentino**
## What's the purpose of this pull request?
This PR tries to take the styling out of the store-ui. The idea is that the styling system stays in stores (storecomponents.store, marinbrasil.store) in such a way that we can be compatible with tailwind, tachyons, stitches, etc. depending on the store which technology you are going to use.
I am thinking of data attributes to describe the behavior of the components.
Also, this PR makes the layout of atomic design and arranges the storybook accordingly
## How it works? 
<em>Tell us the role of the new feature, or component, in its context.</em>

## How to test it?
<em>Describe the steps with bullet points. Is there any external link that can be used to better test it or an example?</em> 

## References
<em>Spread the knowledge: is there any content you used to create this PR that is worth sharing?</em>  

<em>Extra tip: adding references to related issues or mentioning people important to this PR may be good for the documentation and reviewing process</em>

### CODE REVIEW

**Review by tlgimenes — COMMENTED** [review]
for me it's LGTM, however, is it possible to add some guide on how to style the button, doing different states, like disabled etc? I didn't get still 100% of this, thanks!

File: packages/store-ui/package.json [hunk 1]
```diff
@@ -53,8 +53,8 @@
   },
   "peerDependencies": {
     "@vtex/gatsby-plugin-i18n": "^0.280.0",
-    "react": "^0.0.0-experimental-7f28234f8",
-    "react-dom": "^0.0.0-experimental-7f28234f8"
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2"
```
Comment (klzns) [inline packages/store-ui/package.json:57]: Do we need to change the version of React? I think the experimental version has some concurrent mode features that we are using.

Comment (emersonlaurentino) [reply packages/store-ui/package.json:57]: we use this version exact to concurrent mode, but in the theme, on store ui we do not need concurrent mode, and this version broke the storybook runtime.

**Comment by emersonlaurentino** [comment]
> for me it's LGTM, however, is it possible to add some guide on how to style the button, doing different states, like disabled etc? I didn't get still 100% of this, thanks!

@tlgimenes Yes, makes sense to me. I will create some guides on how to style the components.

**Comment by emersonlaurentino** [comment]
@klzns @tlgimenes what do u think about the spread props or use specific props, like onClick, disabled, etc...?

File: packages/store-ui/src/atoms/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import type { FC, ReactNode } from 'react'
```
Comment (igorbrasileiro) [inline packages/store-ui/src/atoms/Button/Button.tsx]: What do you think about changing the filename to index?

File: packages/store-ui/src/atoms/Button/UIButton.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  children,
+  variant = 'sui-button',
+  ...props
+}) => {
+  return (
+    <Button as={RootButton} variant={variant} {...props}>
+      {children}
+    </Button>
+  )
+}
```
Comment (igorbrasileiro) [inline packages/store-ui/src/atoms/Button/UIButton.tsx]: Why this UIButton is needed, can use RootButton directly, don't? Variant props can be used like that 

```tsx
<button sx={{ variant }}>...</button>
```

**Review by igorbrasileiro — COMMENTED** [review]
I'm also asking me about it these questions 😃

File: packages/store-ui/src/atoms/Button/UIButton.tsx (hunk 3 above)
Comment (emersonlaurentino) [reply packages/store-ui/src/atoms/Button/UIButton.tsx]: this component no more exists

File: packages/store-ui/src/atoms/Button/Button.tsx (hunk 2 above)
Comment (emersonlaurentino) [reply packages/store-ui/src/atoms/Button/Button.tsx]: I'll create an index file, thanks!

**Comment by tlgimenes** [comment]
> @klzns @tlgimenes what do u think about the spread props or use specific props, like onClick, disabled, etc...?

I'm not sure. I prefer spreading because it makes our framework more hackeable, however it makes it more difficult to maintain I guess. I'm really not sure, but spreading seems ok to me

**Comment by Gmantiqueira** [comment]
> for me it's LGTM, however, is it possible to add some guide on how to style the button, doing different states, like disabled etc? I didn't get still 100% of this, thanks!

We can also use the Knobs addon for storybook to test the different behaviors of changing props. 
https://storybook.js.org/addons/@storybook/addon-knobs/

**Comment by Gmantiqueira** [comment]
> > @klzns @tlgimenes what do u think about the spread props or use specific props, like onClick, disabled, etc...?
> 
> I'm not sure. I prefer spreading because it makes our framework more hackeable, however it makes it more difficult to maintain I guess. I'm really not sure, but spreading seems ok to me

I think it's better to spread props in an atom component to make it as flexible as possible. And since it's a component that is almost an HTML element (<button>), we need this versatility to use this component like a normal HTML element, with a11y attributes, ids, test-ids, roles, autofocus, etc.

**Comment by lbebber** [comment]
If we want to make it extremely flexible and barebones, why not just use the `<button />` element itself?

This is more or less a rhetorical question, because it might or might not make sense to create a specific `<Button />` element, but I think the answer should help in deciding its API.

**Comment by Gmantiqueira** [comment]
> If we want to make it extremely flexible and barebones, why not just use the `<button />` element itself? This component is basically just that.
> 
> This is more or less a rhetorical question, because it might or might not make sense to create a specific `<Button />` element, but I think the answer should help in deciding its API. What is its purpose?
> 
> Because if it's just an HTML element to have some component semantic/add to a storybook, then yeah, the `...rest` is necessary.

It's nice to make a component for button if we're going to make a base style or base config for every button, eventually. We don't have an styleguide for stores yet but we can have a default style for store's buttons, so it's nice to already have this component like this.

**Comment by igorbrasileiro** [comment]
> It's nice to make a component for button if we're going to make a base style or base config for every button, eventually. We don't have an styleguide for stores yet but we can have a default style for store's buttons, so it's nice to already have this component like this.

🪂  But as far as I know, we won't have styles/themes inside the store-ui. So I agree with @lbebber

**Comment by Gmantiqueira** [comment]
> > It's nice to make a component for button if we're going to make a base style or base config for every button, eventually. We don't have an styleguide for stores yet but we can have a default style for store's buttons, so it's nice to already have this component like this.
> 
> 🪂 But as far as I know, we won't have styles/themes inside the store-ui. So I agree with @lbebber

I don't like this idea about not having any styles inside the store-ui. Some components simply "doesn't work" without a base style, just imagine the Minicart, our Search or the Login UI with the native HTML style. Also, we need a base style to make our guide with the Storybook anyway. 

I prefer to have a default style like storecomponents.store inside store-ui and then overwrite each custom style inside each store. Today we are importing all those default styles inside the store, so it's very confusing to know which one is custom or default.

**Comment by tlgimenes** [comment]
> > for me it's LGTM, however, is it possible to add some guide on how to style the button, doing different states, like disabled etc? I didn't get still 100% of this, thanks!
> 
> We can use the Knobs addon for storybook to test the different behaviors of changing props.
> https://storybook.js.org/addons/@storybook/addon-knobs/

🙏  this is really important I guess

**Pull request merged**